### PR TITLE
#177 show assignment on review sections

### LINKED ISSUE #177: Show assigned to you on the section
Screen Shot 2020-12-14 at 2 28 44 PM

### PR DESCRIPTION
**Pull request by nmadruga**
Fixes #177 

Using [Backend-#177F](https://github.com/openmsupply/application-manager-server/pull/156)
Based on changes from PR #221, please merge that one before merging this one.

### Changes:
- Add `assigned` to store each reviewer assigned to each section on the sectionStructure that is passed to each `ReviewSection`. On the Reviewer page, will pass in the props, if the section is assigned to you or not.
- For Reviewer display on each section if its assigned to you or not - later on for Consolidator should show each Reviewer assigned to each section.
- Show the **Review** button or **edit** icon for responses assigned to the current Reviewer.
- A few layout changes to show the Assignment and review buttons aligned to the right.

TODO: For now the current reviewer is always user id: 6, this will be fixed on another issue to use the current reviewer logged in.

### CODE REVIEW

**Comment by CarlosNZ** [comment]
Just a quick comment before I look at this in more detail.

When I go to a review page, it's showing as all sections assigned to me. Isn't it supposed to show what's assigned to reviewer id 6? In the database it's correctly assigned, I'm just wondering whether this is supposed to be reflected in the front-end, or if that's for another PR?

**Comment by nmadruga** [comment]
> When I go to a review page, it's showing as all sections assigned to me. Isn't it supposed to show what's assigned to reviewer id 6? In the database it's correctly assigned, I'm just wondering whether this is supposed to be reflected in the front-end, or if that's for another PR?

It should be showing on this PR. 
Have you run the good ol' `yarn database_init` right?
Maybe a clean of cache with **log out** and **log in** again?

That's how it should show (other sections):
Screen Shot 2020-12-18 at 12 52 28 PM

File: src/components/Review/ReviewSection.tsx [hunk 1]
```diff
@@ -1,39 +1,58 @@
...
+      style={{ backgroundColor: 'WhiteSmoke', marginLeft: '10%', marginRight: '10%' }}
     >
-      <Header as="h2" content={`${section.title}`} style={{ color: 'Grey' }} />
+      <Grid columns={2}>
+        <Grid.Row>
+          <Grid.Column>
+            <Header as="h2" content={`${section.title}`} style={{ color: 'Grey' }} />
+          </Grid.Column>
+          <Grid.Column>
+            <Container textAlign="right">{getSectionAssignment}</Container>
```
Comment (CarlosNZ) [inline src/components/Review/ReviewSection.tsx:44]: Just something to think about, maybe don't want to change it for now, but -- I kind of think of `get` methods as functions that return a value. I wonder if functions that return visual components (i.e. JSX) might be called `show..` e.g. `showSectionAssignment` (or maybe `displaySectionAssignment). Cos you're really _getting_ the assignment, you're getting the component to _display_ the assignment. Subtle point, see what you think.

I like the general pattern of functions that return JSX though, I'll try and think to use it more. :)

**Review by CarlosNZ — APPROVED** [review]
Great! Just a couple of questions, up to you if you want to address them.

One other thing, not sure if it's related to this issue: I notice in the database that the `review_question_assignment_id` field in review_response is all NULL. We can link from review_response to review_question_assignment via application_response id -> template_id -> review_question_assignemt, so is this field redundant?

File: src/components/Review/ReviewSection.tsx (hunk 1 above)
Comment (nmadruga) [reply src/components/Review/ReviewSection.tsx:44]: No problem I'll change this one.

**Comment by nmadruga** [comment]
> One other thing, not sure if it's related to this issue: I notice in the database that the review_question_assignment_id field in review_response is all NULL. We can link from review_response to review_question_assignment via application_response id -> template_id -> review_question_assignemt, so is this field redundant?

@CarlosNZ I'm not sure if I understood this, sorry! I tried reading a couple of times but just didn't get the message. I think I'll merge this one and will ask you in person so I can understand. Anyway, we can change as part of another issue.

**Pull request merged**